Commands/GenerateWebpImages.php

```diff
@@ -2,22 +2,87 @@
 
 namespace FroshWebP\Commands;
 
+use FroshWebP\Components\ImageStack\Arguments;
+use FroshWebP\Factories\WebpConvertFactory;
+use FroshWebP\Models\WebPMedia;
+use FroshWebP\Repositories\WebPMediaRepository;
 use FroshWebP\Services\WebpEncoderFactory;
 use Shopware\Commands\ShopwareCommand;
+use Shopware\Components\Model\ModelManager;
 use Symfony\Component\Console\Helper\ProgressBar;
 use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
 use Symfony\Component\Console\Output\OutputInterface;
 
 /**
  * Class GenerateWebpImages
  */
 class GenerateWebpImages extends ShopwareCommand
 {
+    /**
+     * @var ModelManager|null
+     */
+    private $modelManager = null;
+    /**
+     * @var WebPMediaRepository|null
+     */
+    private $webpRepository = null;
+
+    /** @var WebpEncoderFactory|null */
+    private $encoderFactory = null;
+
+    /** @var \FroshWebP\Components\WebpEncoderInterface[]|null */
+    private $runnableEncoders = null;
+
+    /** @var |null */
+    private $webpQuality = null;
+
+    /** @var \Shopware\Bundle\MediaBundle\MediaService|null */
+    private $mediaService = null;
+
+    /**
+     * GenerateWebpImages constructor.
+     *
+     * @param ModelManager                              $manager
+     * @param \Shopware\Bundle\MediaBundle\MediaService $mediaService
+     * @param WebpEncoderFactory                        $webpEncoder
+     */
+    public function __construct(ModelManager $manager, \Shopware\Bundle\MediaBundle\MediaService $mediaService, WebpEncoderFactory $webpEncoder, array $webpConfig)
+    {
+        $this->modelManager = $manager;
+        $this->mediaService = $mediaService;
+        $this->webpRepository = $manager->getRepository(WebPMedia::class);
+        $this->encoderFactory = $webpEncoder;
+        $this->runnableEncoders = WebpEncoderFactory::onlyRunnable($this->encoderFactory->getEncoders());
+        $this->webpQuality = $webpConfig['webPQuality'];
+        parent::__construct();
+    }
+
     protected function configure()
     {
         $this
             ->setName('frosh:webp:generate')
-            ->setDescription('Generate webp images for all orginal images');
+            ->setDescription('Generate webp images for all orginal images')
+            ->addOption('stack', 's', InputOption::VALUE_OPTIONAL, 'process amount per iteration')
+            ->addOption('offset', 'o', InputOption::VALUE_OPTIONAL, 'process amount per iteration')
+            ->addOption('force', 'f', InputOption::VALUE_NONE, 'forces recreation')
+            ->addOption('setCollection', 'c', InputOption::VALUE_OPTIONAL | InputOption::VALUE_IS_ARRAY, 'only generates medias for specified collection')
```

Comment:
Maybe add to the command description an example option value

Reply:
Is this [description](https://github.com/FriendsOfShopware/FroshWebP/blob/59ea13d4cefefb20b2e88f4ca2a5d5d523ad1646/Commands/GenerateWebpImages.php#L69) okay?